test/lit.cfg

```diff
@@ -1340,6 +1340,99 @@ elif run_os == 'linux-androideabi' or run_os == 'linux-android':
         '-L%s' % make_path(test_resource_dir, config.target_sdk_name)])
     # The Swift interpreter is not available when targeting Android.
     config.available_features.discard('swift_interpreter')
+elif run_os == 'wasi':
+    tools_directory = pipes.quote(make_path(config.wasi_sdk_path, "bin"))
+
+    if run_cpu == 'wasm32':
+        lit_config.note("Testing WebAssembly/WASI " + config.variant_triple)
+    else:
+        lit_config.fatal("Unknown environment %s %s" % (run_os, run_cpu))
+
+    config.target_object_format = "wasm"
+    config.target_shared_library_prefix = 'lib'
+    config.target_shared_library_suffix = ".a"
+    config.target_sdk_name = "wasi"
+    config.target_runtime = "native"
+
+    # Exclude test cases that use objc-interop because clang doesn't support it
+    # with WebAssembly binary file yet.
+    def use_objc_interop(path):
+      with open(path) as f:
+        return '-enable-objc-interop' in f.read()
```

Comment:
I would rather that we wait for the changes that @gribozavr and @MForster are working on to restructure the tests and disable the directory instead.